### ISSUE TITLE
train the best model we can on contacts-v1 dataset

### ISSUE THREAD

**Issue by timodonnell** [Open-Athena/MarinFold#61]
We have a new dataset here: https://huggingface.co/buckets/open-athena/MarinFold/tree/data/document_structures/contacts_v1

More details about it are [here](https://github.com/Open-Athena/MarinFold/tree/main/marinfold/marinfold/document_structures/contacts_v1).

The main difference from our previous contacts-and-distances-v1 is we have gotten rid of distance statements (our evals will be entirely on contact recapitulation).

A few other notes:
1) Documents are a lot shorter. Only a handful make it all the way to 8192 tokens. The total train corpus is only 4.6B tokens now (vs 40B in contacts-and-distances-v1). I think document packing is probably worthwhile here assuming it is possible only train on only prefixes of documents (rather than suffixes). (If this is too few tokens I can curate more data if needed.)
1) We are defining contacts more carefully now (using [pyconfind](https://github.com/timodonnell/pyconfind)) so they hopefully each carry more information 
1) I am hoping this will train fine without any loss masking
1) Amino acid sequences are now specified in random order (pairs of position, aa) as a kind of poor man's masked language modeling objective. I should test the impact of this at some point.
1) Based on @eric-czech 's findings, we are now excluding proteins with foldseek structural clusters that have only 1 or 2 members from the training set. Due to this we now have 4.21M documents rather than 5.39M.

@eric-czech Are you up for handling the model training here?

**Comment by eric-czech** [comment]
I am, yea.  One big tuning sweep (at full scale) inclusive of epochs and regularization would actually make sense here if we take the model size and dataset as fixed.  That's a huge assumption though, so why 1.5B params?  Going bigger will almost certainly be better by loss alone.  Should we revisit this or take 1.5B as given?

**Comment by timodonnell** [comment]
No need to take the model size as fixed. For me the goal is just to get as good a model as possible in a reasonable amount of time (like a week or two). If you think bigger will help I am all for it.

The reason I suggested 1.5B is that 1.5B is currently the best model I've trained on the earlier dataset (contacts-and-distances-v1). In that single-epoch setting the 3B is better than 1B but worse than the 1.5B:

<img width="610" height="293" alt="Image" src="https://github.com/user-attachments/assets/a0a9466b-947c-4438-a190-1833235e2619" />

But makes sense that your optimized settings may change this. Also it does seem like epoching will help - the 1B restart above is on track to start beating the 1.5B soon.

**Comment by eric-czech** [comment]
Alright, I'll plan on running https://github.com/Open-Athena/MarinFold/issues/75 at full scale then until I stop seeing improvements being made.  If that seems to be improving on your vanilla 1.5B setup, I'll repeat it at different model scales.  That's more or less how how the method mentioned in that issue works (from @kothasuhas).

**Comment by eric-czech** [comment]
The sweep from #75 is running now ([wandb](https://api.wandb.ai/links/eric-czech/9t0qj6mf)) and optimizes LR/WD (Qwen3 1.5B + AdamW) for epochs in `[1, 2, 4, 8]`.  They're running in order so the 1-4 epoch results should finish later next week and make it possible to interpolate a comparison to the 2.7 epoch results from https://github.com/Open-Athena/MarinFold/pull/70.  The 8 epoch results are set to finish ~5 days later and hopefully they make for a significant improvement 🤞.

**Comment by eric-czech** [comment]
The speedup from this (in tokens) is looking better than I would have guessed.  Some of the 2 epoch runs already finished with loss below the 2.7 epoch [baseline](https://github.com/Open-Athena/MarinFold/issues/67#issuecomment-4691754127) (> 1.35x speedup):

<img width="3555" height="1513" alt="Image" src="https://github.com/user-attachments/assets/7f27d9cd-621b-4b4f-ad1c-9e60544e9b2c" />

The only reason these are completing so quickly is that I'm seeing excellent TRC throughput right now on bigger slices, e.g. `v5p-64` is running ~6x faster than `v5p-8`/`v6e-4` on this data with global batch 128:

<p align="center">
<img width="448" alt="Image" src="https://github.com/user-attachments/assets/9c04a56c-ff83-4fb4-a53f-4c4d9cf0471b" />
</p>

- `tok/s (total)`: Tokens processed divided by wall time since the beginning of training; inclusive of delays from slice availability, preemptions, iris scheduling, intermittent failures, multi-host overhead, etc. 
- `tok/s (train)`: Token throughput during training per `throughput/tokens_per_second` from levanter

Normally, the training throughput on those slices is a good bit higher but the sort of "all-in" throughput (`tok/s (total)`) frequently doesn't seem that much better (summer break + bug fixes I guess 🎉). 

FWIW @timodonnell, I'm using logic like [`skills/schedule-sweep.md`](https://github.com/eric-czech/marin-agent-kb/blob/main/skills/schedule-sweep.md) to be able to keep 10-20 runs going, with Claude tracking true throughput and making smarter decisions on what TPU slices are best over time.  I'm seeing `v5p-32/64` and `v6e-16` are great choices ATM if you kick off any more single runs in the near future.

**Comment by eric-czech** [comment]
🤖 Refreshed: final validation loss vs run completion time, colored by epoch count, with the lower-loss best-so-far frontier. More epochs + higher LR push loss down; best so far ~2.76 at 8 epochs / lr=1e-3, below the 1.5B baseline (2.98).

<img width="3555" height="1513" alt="Image" src="https://github.com/user-attachments/assets/3c32bc18-9769-4af0-b93e-f95f6775f99c" />

**UPDATE**: Final results are at https://github.com/Open-Athena/MarinFold/issues/75#issuecomment-4916194271; no improvement on mid-sweep update above.
